packages/govern-console/src/lib/web3-contracts.ts

```diff
@@ -58,32 +58,31 @@ export function useContractReadOnly(
   const ethEndpoint = getNetworkNode()
   const [, abi] = getKnownContract(name)
 
-  const ethProvider = useMemo(
-    () => (ethEndpoint ? new providers.JsonRpcProvider(ethEndpoint) : null),
-    [ethEndpoint],
-  )
-
-  return useMemo(
-    () => (address ? new EthersContract(address, abi, ethProvider) : null),
-    [abi, address, ethProvider],
-  )
+  return useMemo(() => {
+    if (!address) {
+      return null
+    }
+    return new EthersContract(
+      address,
+      abi,
+      ethEndpoint ? new providers.JsonRpcProvider(ethEndpoint) : undefined,
+    )
+  }, [abi, address, ethEndpoint])
 }
 
-export function useTokenBalance(
-  symbol: string,
-  address = '',
-): EthersUtils.BigNumber {
-  const { account } = useWallet()
+export function useTokenBalance(symbol: string, address = ''): EthersBigNumber {
   const [balance, setBalance] = useState(bigNum(-1))
   const tokenContract = useKnownContract(`TOKEN_${symbol}`)
+  const { wallet } = useWallet()
+  const { account } = wallet
 
-  const cancelBalanceUpdate = useRef(null)
+  const cancelBalanceUpdate = useRef<(() => void) | null>(null)
 
   const updateBalance = useCallback(() => {
     let cancelled = false
 
     if (cancelBalanceUpdate.current) {
-      cancelBalanceUpdate.current()
+      cancelBalanceUpdate.current?.()
```

Comment:
I’m not sure it was happening, but before this change it was always calling `cancelBalanceUpdate()` despite the possibility of it being `null`.

Reply:
Hmm and can it be null? As we're already checking if the ref is not `falsy` on the `if` statement.

Reply:
True haha, sorry I was having my type-only glasses. I guess we could remove the condition now :smile:

Reply:
Haha yes!

(⌐■_■)
( •_•)>⌐■-■
(•_•)

Reply:
Done in https://github.com/aragon/govern/pull/261/commits/83b0b02f27887c0ee36f7332b27f30a1cef9fac2

(I also moved the `cancelBalanceUpdate.current = null`)